### PR TITLE
feat: add oscd-select based on wizard-select

### LINKED ISSUE #3: Create `oscd-select` based on `wizard-select`
As a plugin author and maintainer of already existing plugins, I need to have `open-scd`s `wizard-select` available through nmp.

**Requirements:**
- move `wizard-select` to `oscd-select` repo
- change tag name to `oscd-select`
- do !NOT! extend from `mwc-select` but use it in the `render` method. This allows us to exchange the now used material web components with another web-component library.
- add properties

| name | type | description |
|---|---|---|
| value | string \| null | set and get Elements attribute value null means attribute is missing |
| disabled | boolean | disables both the input field and the null switch |
| defaultValue | string | is indicating the default value if `maybeValue` is null |
| label | string | pass to mwc-select |
| required | boolean | pass to input field |
| nullable | boolean | Whether null is valid `maybeValue` output |
| helper | string | pass to input field |
| validationMessage | string | pass to input field helper if input is invalid |
| selected | ListItemBase\|null | Selected list item element. | 
| items | ListItemBase[] | List of selectable items. |
| index | number | Index of selected list item. | 

- expose methods

| name | type | description |
| --- | --- | --- |
| checkValidity | boolean | Whether the select field is valid |

- propagate event 
  `change` event for any changes on `value`

- add screenshot tests
- add storybook

**Additional Information**
This web-component is used pretty much everywhere in the software

### PR DESCRIPTION
**Pull request by Stef3st**
closes #3

### CODE REVIEW

**Review by juancho0202 — APPROVED** [review]
I think this meets our [component guide](https://github.com/openscd/.github/blob/23491bd2408f3b7edaa8b0149a3d57aca1adaf40/docs/component_guide.md). You can merge with or without my CEM comments suggestions.

File: components/oscd-select/src/oscd-select.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+
+  const dispatched = element.dispatchEvent(copy);
+  if (!dispatched) {
+    event.preventDefault();
+  }
+  return dispatched;
+}
+
+/**
+ * @tag oscd-select
```
Comment (pascalwilbrink) [inline components/oscd-select/src/oscd-select.ts:36]: none of the props are documented

File: components/oscd-select/src/oscd-select.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+      this.null = false;
+      this.value = value;
+    }
+  }
+  /** The default `value` displayed if [[`maybeValue`]] is `null`. */
+  @property({ type: String })
+  defaultValue = "";
+  /** Additional values that cause validation to fail. */
+  @property({ type: Array })
+  reservedValues: string[] = [];
```
Comment (pascalwilbrink) [inline components/oscd-select/src/oscd-select.ts:75]: not used

**Review by pascalwilbrink — CHANGES_REQUESTED** [review]

File: components/oscd-select/src/oscd-select.ts (hunk 1 above)
Comment (Stef3st) [reply components/oscd-select/src/oscd-select.ts:36]: They are documented, but above the properties themselves. The CEM analyzer will get these aswell.

File: components/oscd-select/src/oscd-select.ts (hunk 2 above)
Comment (Stef3st) [reply components/oscd-select/src/oscd-select.ts:75]: This is copied from the wizard-select. Will this never be used? Then Im ok with removing the property

**Review by juancho0202 — APPROVED** [review]
Annotations look good and the removed property is the same as the one that @pascalwilbrink pointed out

**Pull request merged**